main.tf

```diff
@@ -0,0 +1,314 @@
+    storage  = "storage.googleapis.com/${local.destination_name}"
+  }
+
+  # API map
+  api_map = {
+    pubsub   = "pubsub.googlapis.com"
+    bigquery = "bigquery-json.googleapis.com"
+    storage  = "storage.googleapis.com"
+  }
+
+  # Destination console URLs based on the sink destination type
+  web_map = {
+    pubsub   = "https://console.cloud.google.com/cloudpubsub/topics/${local.destination_name}?project=${local.destination_project}"
+    bigquery = "https://bigquery.cloud.google.com/dataset/${local.destination_project}:${local.destination_name}"
+    storage  = "https://console.cloud.google.com/storage/browser/${local.destination_name}?project=${local.destination_project}"
+  }
+
+  # Sink output data
+  sink_output = {
+    resource_name = "${local.is_project_level ? var.project : local.is_folder_level ? var.folder : local.is_org_level ? var.org_id : ""}"
+    resource      = "${local.is_project_level ? element(concat(google_logging_project_sink.sink.*.id, list("")), 0) : local.is_folder_level ? element(concat(google_logging_folder_sink.sink.*.id, list("")), 0) : local.is_org_level ? element(concat(google_logging_organization_sink.sink.*.id, list("")), 0) : local.is_billing_level ? element(concat(google_logging_billing_account_sink.sink.*.id, list("")), 0) : ""}"
+    writer        = "${local.is_project_level ? element(concat(google_logging_project_sink.sink.*.writer_identity, list("")), 0) : local.is_folder_level ? element(concat(google_logging_folder_sink.sink.*.writer_identity, list("")), 0) : local.is_org_level ? element(concat(google_logging_organization_sink.sink.*.writer_identity, list("")), 0) : local.is_billing_level ? element(concat(google_logging_billing_account_sink.sink.*.writer_identity, list("")), 0) : ""}"
+    console_link  = "${local.is_project_level ? "https://console.cloud.google.com/logs/exports?project=${var.project}" : ""}"
+    name          = "${var.name}"
+    filter        = "${var.filter}"
+  }
+
+  # Sink destination output data
+  destination_output = {
+    resource     = "${local.is_pubsub ? element(concat(google_pubsub_topic.topic.*.id, list("")), 0) : local.is_bigquery ? element(concat(google_bigquery_dataset.dataset.*.id, list("")), 0) : local.is_storage ? element(concat(google_storage_bucket.bucket.*.id, list("")), 0) : ""}"
+    self_link    = "${local.destination_uri}"
+    console_link = "${local.destination_console_link}"
+    type         = "${local.destination_type}"
+    project      = "${local.is_pubsub ? element(concat(google_pubsub_topic.topic.*.project, list("")), 0) : local.is_bigquery ? element(concat(google_bigquery_dataset.dataset.*.project, list("")), 0) : local.is_storage ? element(concat(google_storage_bucket.bucket.*.project, list("")), 0) : ""}"
+  }
+}
+
+#---------------------#
+# Logsink destination #
+#---------------------#
+
+# Enable destination API
+resource "null_resource" "enable_destination_api" {
```

Comment:
Can we look into doing this natively in Terraform using https://www.terraform.io/docs/providers/google/r/google_project_service.html with `disable_on_destroy` set to false?

Reply:
Sure, just noticed there was a typo in 'pubsub.googlapis.com' --> 'pubsub.googleapis.com' and that's why when I tried to use that resources, things failed. Will correct.